From b94abb1cf374ed624360d9aff1af01e987685bfe Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Wed, 30 Oct 2019 13:40:36 +0800
Subject: [PATCH 45/53] bmq: [Sync] 0ff7b2cfbae3 tasks, sched/core: Ensure
 tasks are available for a grace period after leaving the runqueue

---
 kernel/sched/bmq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/bmq.c b/kernel/sched/bmq.c
index 24d741697cf1..eecd5ddec6db 100644
--- a/kernel/sched/bmq.c
+++ b/kernel/sched/bmq.c
@@ -2182,7 +2182,7 @@ static struct rq *finish_task_switch(struct task_struct *prev)
 		/* Task is done with its stack. */
 		put_task_stack(prev);
 
-		put_task_struct(prev);
+		put_task_struct_rcu_user(prev);
 	}
 
 	tick_nohz_task_switch();
-- 
2.24.0.155.gd9f6f3b619

